state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1796,21 +1796,21 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			IngressOrEgress::Egress => T::ChainTracking::estimate_egress_fee(asset),
 		};
 
-		let transaction_fee = if asset == <T::TargetChain as Chain>::GAS_ASSET {
+		let fees_withheld = if asset == <T::TargetChain as Chain>::GAS_ASSET {
 			// No need to schedule a swap for gas, it's already in the gas asset.
 			Self::accrue_withheld_fee(asset, sp_std::cmp::min(fee_estimate, available_amount));
 			fee_estimate
```

Comment:
you need to do min here as well, right? @dan

Reply:
I added the min in the transaction fee in the `else` branch. I figured even if we swallow the deposit, we still want to convert the input to our gas asset (because we might need to pay to fetch it).

Reply:
(hope that makes sense...)